### DISCUSSION TITLE
Build Java binding for Windows

### DISCUSSION THREAD

**Discussion by Baerlie** [coqui-ai/STT#2346] [Q&A]
Hi,

I'm currently building a Java client that uses Coqui STT but found out that the libstt-jni.so included in the release files seems to be a Linux library. Java can't load it (and it would probably be a dll if it were built for Windows). I checked the release files for a Windows JNI library, but were not able to find one.

Is there some info in the documentation about building the stt-jni and stt bindings for Java for Windows? Or does the command below also build the Java bindings on/for Windows? Can this command also be used on Linux to build Windows libs?

```
bazel build --workspace_status_command="bash native_client/bazel_workspace_status_cmd.sh" -c opt --copt=/arch:AVX --copt=/arch:AVX2 //native_client:libstt.so
```

Thanks & br
Beatrice

**Comment by Baerlie** [comment]
Figured out by myself! 
I will write a summary about what I did later.

**Comment by HeribertoCristofaro7vB** [comment]
Unfortunately, as of the latest release of Coqui STT (v0.10.0), there is no pre-built JNI library for Windows included in the release files. However, you can build the JNI library yourself for Windows using the same command you mentioned, with the addition of the --host_javabase flag pointing to the path of a JDK on your Windows machine. Here is an example command:
bazel build --host_javabase=C:/Program Files/Java/jdk-11.0.13 --workspace_status_command="bash native_client/bazel_workspace_status_cmd.sh" -c opt //native_client:libstt_jni.dll

Note that the --host_javabase flag is necessary for building JNI libraries on Windows, as it tells Bazel where to find the JDK on your machine.

As for building the Java bindings, you can build them using the following command:
bazel build //java:libdeepspeech.so

This command builds the Java bindings as a shared library (libdeepspeech.so) that can be loaded by your Java client. However, note that the libdeepspeech.so library requires the libdeepspeech.so native library to be present on the system, so you'll need to make sure that both libraries are available on your Windows machine.

I hope this helps! Let me know if you have any further questions.
